hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/event/Event.java

```diff
@@ -32,4 +32,14 @@ public interface Event<TYPE extends Enum<TYPE>> {
   TYPE getType();
   long getTimestamp();
   String toString();
+
+  /**
+   * In case of parallel execution of events in the same dispatcher,
+   * the result of this method will be used as semaphore.
+   * If method returns null, then a default semaphore will be used.
+   * @return the semaphore
+   */
+  default String getLockKey() {
+    return null;
```

Comment:
Is returning null expected here?

Reply:
Hi @slfan1989 !

Thanks for the review.
Yes, that is expected. 
If we dont specify lockKey for an event we should return with null, so these events will be executed in sequential not parallel. The method is used in the MultiDispatcherLocks.